src/main/java/lotto/domain/LottoStore.java

```diff
@@ -0,0 +1,51 @@
+package lotto.domain;
+
+import static lotto.constant.DomainConstant.LOTTO_PRICE;
+import static lotto.constant.DomainConstant.START_INDEX;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import lotto.constant.ErrorMessage;
+
+public final class LottoStore {
+    private final NumberGenerator numberGenerator;
+
+    private LottoStore(NumberGenerator numberGenerator) {
+        this.numberGenerator = numberGenerator;
+    }
+
+    public static LottoStore from(NumberGenerator numberGenerator) {
+        return new LottoStore(numberGenerator);
+    }
+
+    public List<Lotto> purchaseLottos(long investmentAmount) {
+        validateInvestmentAmount(investmentAmount);
+        int numberOfLottos = calculateNumberOfLottos(investmentAmount);
+        return createLottos(numberOfLottos);
+    }
+
+    private List<Lotto> createLottos(int numberOfLottos) {
+        return IntStream.range(START_INDEX.getValue(), numberOfLottos)
+            .mapToObj(i -> new Lotto(createLottoNumbers()))
+            .collect(Collectors.toList());
+    }
```

Comment:
(이렇게 바꿀 수 있어요!)

```suggestion
        return Stream.generate(this::createLottoNumbers)
                .limit(numberOfLottos)
                .toList();
    }
```
limit을 통해서 구매 개수만큼 생성을 제한하면 조금더 직관적으로 표현 가능해요!

Reply:
매번 리뷰 감사합니다! :) Stream에 대해서 조금 더 찾아봐야겠네요